errutil/error.go

```diff
@@ -16,3 +18,52 @@ type Error string
 func (e Error) Error() string {
 	return string(e)
 }
+
+// Chain creates a chain of errors suitable to be used
+// with Go's Unwrap interface through functions like
+// errors.Is and errors.As.
+// Chaining order will be the same as the order of the
+// arguments, the first error is the head wrapping up
+// the next one, and so goes on.
+//
+// An empty list of errors will return a nil error.
+func Chain(errs ...error) error {
+	// TODO(katcipis): should we do something when
```

Comment:
@i4ki WDYT about this ? if the errs slice has a nil in the middle it will disrupt the chain (current untested behavior). If we keep it this way I would make it explicit/test it...but Im not sure x_x. Lets say an slice only has nils inside...should it return a non-nil error ? (right now it will, with a nil head/tail).

Reply:
yes, I think we should filter nil errs out and if no err is left then return nil.

Reply:
Will add this on another PR 🚀